Userland/Libraries/LibCore/Resource.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <AK/Error.h>
+#include <AK/MemoryStream.h>
+#include <AK/OwnPtr.h>
+#include <AK/Span.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Variant.h>
+#include <LibCore/File.h>
+#include <LibCore/MappedFile.h>
+
+namespace Core {
+class Resource {
+public:
+    static ErrorOr<NonnullOwnPtr<Resource>> load_from_uri(StringView);
+
+    [[nodiscard]] bool is_file() const { return !m_data.has<DirectoryTag>(); }
+    [[nodiscard]] bool is_directory() const { return m_data.has<DirectoryTag>(); }
+
+    [[nodiscard]] String uri() const;
```

Comment:
URL probably?

Reply:
I don't want a URL though. I want a URI. https://www.rfc-editor.org/rfc/rfc3986

Reply:
AK::URL should be URI-compatible. If not, leave a FIXME?

Reply:
I don't want to use AK::URL from LibCore because we might move URL to its own library to allow use of LibUnicode. We can't have that kind of circular dependency in LibCore.